File: day_1/js/main.js
const fs = require("fs");

console.log(
  "Calories MAX",
  fs
    .readFileSync("input.txt", { encoding: "utf-8" })
    .split("\n")
    .reduce(
      (acc, line) => {
        if (!line.length) {
          acc.push([]);
        } else {
          acc[acc.length - 1].push(parseInt(line));
        }
        return acc;
      },
      [[]]
    )
    .map(
      (bucket) =>
        bucket.length ? bucket.reduce((sum, entry) => sum + entry) : 0,
      0
    )
    .reduce((max, entry) => Math.max(max, entry), 0)
);

File: day_1/cpp/main.cpp
#include <iostream>
#include <fstream>

int main(int argc, char *argv[])
{
    using namespace std;

    int latest_max_calories = 0;
    int max_calories = 0;

    ifstream input_file("input.txt");
    string line;
    while (!getline(input_file, line).eof())
    {
        if (line.empty())
        {
            max_calories = max(latest_max_calories, max_calories);
            latest_max_calories = 0;
            continue;
        }

        latest_max_calories += stol(line);
    }

    max_calories = max(latest_max_calories, max_calories);

    cout << "Max calories: " << max_calories << endl;
}